native/core/src/execution/planner.rs

```diff
@@ -3331,4 +3347,146 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_nested_types_extract_missing_struct_names() -> Result<(), DataFusionError> {
```

Comment:
I'm planning to refactor rust planner tests into separate file and factor out common parts, like test parquet creation, test preparation, etc